### PR TITLE
HAWQ-51. Support Kerberos for libyarn

### PR DESCRIPTION
**Pull request by linwen**
codes about supporting Kerberos for libyarn, please review.

### CODE REVIEW

File: depends/libyarn/src/libyarnclient/LibYarnClient.cpp [hunk 1]
```diff
@@ -129,16 +129,16 @@ int LibYarnClient::createJob(string &jobName, string &queue,string &jobId) {
...
 
         //3. wait util AM is ACCEPTED and return the AMRMToken
         ApplicationReport report;
         while (true) {
-            report = appClientAlias->getApplicationReport(appId);
+            report = applicationClient->getApplicationReport(appId);
             LOG(INFO,"LibYarnClient::createJob, appId[cluster_timestamp:%lld,id:%d], appState:%d",
                     appId.getClusterTimestamp(), appId.getId(), report.getYarnApplicationState());
-            if ((report.getAMRMToken().getPassword() != "") && report.getYarnApplicationState() == YarnApplicationState::ACCEPTED) {
+			if ((report.getAMRMToken().getPassword() != "") && report.getYarnApplicationState() == YarnApplicationState::ACCEPTED) {
```
Comment (wangzw) [inline depends/libyarn/src/libyarnclient/LibYarnClient.cpp]: Please do not mix space and tab. Space is preferred.

File: depends/libyarn/src/libyarnclient/LibYarnClient.cpp [hunk 2]
```diff
@@ -150,15 +150,23 @@ int LibYarnClient::createJob(string &jobName, string &queue,string &jobId) {
...
-        Yarn::Token anotherToken;
-        anotherToken.setIdentifier(token.getIdentifier());
-        anotherToken.setKind(token.getKind());
-        anotherToken.setPassword(token.getPassword());
-        anotherToken.setService(token.getService());
-
-        //LOG(INFO,"%s",token.getIdentifier());
-        user.addToken(anotherToken);
+        UserInfo user;
+		if (applicationClient->getMethod() == SIMPLE)
```
Comment (wangzw) [inline depends/libyarn/src/libyarnclient/LibYarnClient.cpp]: Please do not mix space and tab. Space is preferred.

Comment (linwen) [inline depends/libyarn/src/libyarnclient/LibYarnClient.cpp]: Accepted. Thanks.

File: depends/libyarn/src/libyarnclient/LibYarnClient.cpp (hunk 1 above)
Comment (linwen) [inline depends/libyarn/src/libyarnclient/LibYarnClient.cpp]: accepted.

**Comment by wangzw** [comment]
Code is almost good except the space issue. +1

File: depends/libyarn/src/rpc/SaslClient.cpp [hunk 3]
```diff
@@ -122,6 +122,9 @@ std::string SaslClient::evaluateChallenge(const std::string & challenge) {
             free(output);
```
Comment (jiny2) [inline depends/libyarn/src/rpc/SaslClient.cpp]: Seems this in this line, output is always NOT NULL.

**Pull request closed without merging**